src/core/forces_inline.hpp

```diff
@@ -192,36 +193,23 @@ inline Utils::Vector3d calc_non_bonded_pair_force_parts(
 #ifdef GAY_BERNE
   // The gb force function isn't inlined, probably due to its size
   if (dist < ia_params.gay_berne.cut) {
-    auto const forces =
-        gb_pair_force(p1.r.calc_director(), p2.r.calc_director(), ia_params, d,
-                      dist, torque1, torque2);
-    force += std::get<0>(forces);
-    if (torque1) {
-      *torque1 += std::get<1>(forces);
-    }
-    if (torque2) {
-      *torque2 += std::get<2>(forces);
-    }
+    pf += gb_pair_force(p1.r.calc_director(), p2.r.calc_director(), ia_params,
+                        d, dist);
   }
 #endif
-  force += force_factor * d;
-  return force;
-}
-
-inline Utils::Vector3d calc_non_bonded_pair_force(
-    Particle const &p1, Particle const &p2, IA_parameters const &ia_params,
-    Utils::Vector3d const &d, double dist, Utils::Vector3d *torque1 = nullptr,
-    Utils::Vector3d *torque2 = nullptr) {
-  return calc_non_bonded_pair_force_parts(p1, p2, ia_params, d, dist, torque1,
-                                          torque2);
+  pf.f += force_factor * d;
+  return pf;
 }
 
-inline Utils::Vector3d calc_non_bonded_pair_force(Particle const &p1,
-                                                  Particle const &p2,
-                                                  Utils::Vector3d const &d,
-                                                  double dist) {
-  IA_parameters const &ia_params = *get_ia_param(p1.p.type, p2.p.type);
-  return calc_non_bonded_pair_force(p1, p2, ia_params, d, dist);
+inline Utils::Vector3d calc_opposing_torque(ParticleForce const &pf,
+                                            Utils::Vector3d const &d) {
+#ifdef ROTATION
+  // guard against numerical instability in the vector product
```

Comment:
Can you explain what numerical instability you mean? Isn't the vector product just a sum of products some of the components would just vanish if there are zero-components in the torque?

Reply:
The torque isn't part of the vector product, so we cannot rely on vanishing components. If the first torque is a null vector, the second torque has to be a null vector too, because the force is collinear to the distance vector, but due to precision errors there is sometimes a residual with magnitude 1e-16 in the calculated counter-torque. This causes collision detection to crash:
```
 13/158 Test  #93: collision_detection .......................................***Failed   18.87 sec
f=[4.986014e+02   4.449108e+02 -5.791557e+02] d=[ 5.224425e-02  4.661846e-02 -6.068486e-02] t2=[-0.000000e+00 -3.552714e-15 -0.000000e+00]
f=[-1.246381e+02 -2.226122e+02 -1.175642e+02] d=[-4.399725e-02 -7.858208e-02 -4.150015e-02] t2=[-1.776357e-15 -0.000000e+00 -0.000000e+00]
f=[5.373112e+00  -5.428668e+00 -7.974942e+00] d=[ 5.381213e-02 -5.436853e-02 -7.986966e-02] t2=[-0.000000e+00 -5.551115e-17 -0.000000e+00]
terminate called after throwing an instance of 'std::runtime_error'
  what():  misplaced particle with id 200
[seraue:2333958] *** Process received signal ***
[seraue:2333958] Signal: Aborted (6)
[seraue:2333958] Signal code:  (-6)
...
[seraue:2333958] [ 7] /work/jgrad/espresso/build/src/core/EspressoCore.so(CellStructure::check_particle_sorting()+0x167)[0x14a3878f4307]
[seraue:2333958] [ 8] /work/jgrad/espresso/build/src/core/EspressoCore.so(cells_update_ghosts(unsigned int)+0xc1)[0x14a38783b4b1]
[seraue:2333958] [ 9] /work/jgrad/espresso/build/src/core/EspressoCore.so(handle_collisions()+0x819)[0x14a387845989]
[seraue:2333958] [10] /work/jgrad/espresso/build/src/core/EspressoCore.so(integrate(int, int)+0x1c3)[0x14a387892a63]
```

Reply:
Ok I see. I'd say this is more a problem with the coldet test, and should be fixed there.

Reply:
The comment line is misleading, I'll change it.